src/ssh/azext_ssh/rdp_utils.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import platform
+import subprocess
+import time
+import psutil
+
+import colorama
+from colorama import Fore
+from colorama import Style
+
+from knack import log
+
+from azure.cli.core import azclierror
+from azure.cli.core import telemetry
+from . import ssh_utils
+from . import connectivity_utils
+from . import constants as const
+from . import file_utils
+
+logger = log.get_logger(__name__)
+
+
+def start_rdp_connection(ssh_info, delete_keys, delete_cert):
+    try:
+        ssh_process = None
+        log_list = []
+        print_ssh_logs = False
+        ssh_success = False
+
+        resource_port = 3389
+        local_port = _get_open_port()
+
+        while not is_local_port_open(local_port):
+            local_port = _get_open_port()
+
+        if ssh_info.ssh_args is None:
+            ssh_info.ssh_args = ['-L', f"{local_port}:localhost:{resource_port}", "-N"]
+        else:
+            ssh_info.ssh_args = ['-L', f"{local_port}:localhost:{resource_port}", "-N"] + ssh_info.ssh_args
+
+        ssh_process, print_ssh_logs = start_ssh_tunnel(ssh_info)
+        ssh_connection_t0 = time.time()
+        ssh_success, log_list = wait_for_ssh_connection(ssh_process, print_ssh_logs)
+        ssh_utils.do_cleanup(delete_keys, delete_cert, ssh_info.cert_file, ssh_info.private_key_file,
+                             ssh_info.public_key_file)
+        if ssh_success and ssh_process.poll() is None:
+            call_rdp(local_port)
+
+    finally:
+        if ssh_success:
+            ssh_connection_data = {'Context.Default.AzureCLI.SSHConnectionDurationInMinutes':
+                                   (time.time() - ssh_connection_t0) / 60}
+            ssh_connection_data['Context.Default.AzureCLI.SSHConnectionStatus'] = "Success"
+            telemetry.add_extension_event('ssh', ssh_connection_data)
+
+        terminate_ssh(ssh_process, log_list, print_ssh_logs)
+        ssh_utils.do_cleanup(delete_keys, delete_cert, ssh_info.cert_file, ssh_info.private_key_file,
+                             ssh_info.public_key_file)
+        if delete_keys:
+            # This is only true if keys were generated, so they must be in a temp folder.
+            temp_dir = os.path.dirname(ssh_info.cert_file)
+            file_utils.delete_folder(temp_dir, f"Couldn't delete temporary folder {temp_dir}", True)
+
+
+def call_rdp(local_port):
+    from . import _process_helper
+    if platform.system() == 'Windows':
+        colorama.init()
+        print(Fore.GREEN + "Launching Remote Desktop Connection" + Style.RESET_ALL)
+        print(Fore.YELLOW + "To close this session, close the Remote Desktop Connection window." + Style.RESET_ALL)
```

Comment:
It is recommended to use the CLI unified color framework. For specific code samples, please see [code link](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/util/custom.py#L228)
@jiasli If I have something missing, please help supplement it